libs/langgraph/src/tests/pregel.test.ts

```diff
@@ -2094,12 +2173,50 @@ describe("StateGraph", () => {
 
   type Step = [AgentAction | AgentFinish, string];
 
+  type MyContextModel = {
+    session: AxiosInstance;
+    something_else: string;
+  };
+
+  let setup = jest.fn();
+  let teardown = jest.fn();
+
+  async function assertCtxOnce<T>(fn: () => Promise<T>): Promise<T> {
+    expect(setup).toHaveBeenCalledTimes(0);
+    expect(teardown).toHaveBeenCalledTimes(0);
+    try {
+        return await fn();
+    } finally {
+        expect(setup).toHaveBeenCalledTimes(1);
+        expect(teardown).toHaveBeenCalledTimes(1);
+        setup.mockReset();
+        teardown.mockReset();
+    }
+  }
+
+  // Make context function with async context management
+  async function* makeContext(config: Record<string, unknown>): AsyncGenerator<MyContextModel> {
+    if (typeof config !== "object" || config === null) {
+        throw new Error("Config must be an object");
+    }
+    
+    setup();
+    const session = axios.create();
+    try {
+        yield { session, something_else: "hello" };
+    } finally {
+        await session.request({ method: "CLOSE" }); // Simulating axios closing (adjust as needed)
+        teardown();
+    }
+  }
+
   const AgentAnnotation = Annotation.Root({
     input: Annotation<string>,
     agentOutcome: Annotation<AgentAction | AgentFinish | undefined>,
     steps: Annotation<Step[]>({
       reducer: (x: Step[], y: Step[]) => x.concat(y),
     }),
+    context: Annotation<MyContextModel>(Context(makeContext)),
```

Comment:
I think I would make this use the `Context` constructor directly rather than a function

It is nice to not do this for `Annotation` so that it more closely mimics TypeScript type declarations but for things that you're actually passing in I'd say constructor is more intuitive